hazelcast/src/test/java/com/hazelcast/client/topic/ClientReliableTopicTest.java

```diff
@@ -56,7 +56,7 @@
 
 @RunWith(HazelcastParallelClassRunner.class)
 @Category({QuickTest.class, ParallelJVMTest.class})
-public class ClientReliableTopicTest extends HazelcastTestSupport {
+public class ClientReliableTopicTest extends TopicTest {
```

Comment:
Haven't looked at the tests too closely (will look after other comments are addressed) but as a general idea since all implementations share the same interface, I suggest that you write the test once and just parametrize it - either use the reliable topic or regular topic, the ordered or non-ordered topic.

Reply:
As @sancar mentioned, it is the easiest way to extend this class. 
https://github.com/hazelcast/hazelcast/pull/16946#discussion_r421273931

Or do you have any other problem in mind?

Reply:
Yeah, that'll work too. Parametrized tests are good if you stick to a single instance type (member or client) and then want to test different implementations - ordered vs non-ordered, reliable topic vs non-reliable topic.

And then, once you have such a parametrized test using a member as the test instance, you override it as you say and call the same methods on the client.

Anyway, if you see yourself copying the same tests for testing different topic implementations, think about parametrizing tests.